controllers/telemetry/logpipeline_controller.go

```diff
@@ -168,3 +140,79 @@ func (r *LogPipelineController) SetupWithManager(mgr ctrl.Manager) error {
 
 	return b.Complete(r)
 }
+
+func configureFluentBitReconciler(client client.Client, config LogPipelineControllerConfig, flowHealthProber *prober.LogPipelineProber) (*logpipelinefluentbit.Reconciler, error) {
+	fbConfig := logpipelinefluentbit.NewConfig(
+		fbBaseName,
+		config.TelemetryNamespace,
+		logpipelinefluentbit.WithFluentBitImage(config.FluentBitImage),
+		logpipelinefluentbit.WithExporterImage(config.ExporterImage),
+		logpipelinefluentbit.WithPriorityClassName(config.FluentBitPriorityClassName),
+	)
+
+	pipelineValidator := &logpipelinefluentbit.Validator{
+		EndpointValidator:  &endpoint.Validator{Client: client},
+		TLSCertValidator:   tlscert.New(client),
+		SecretRefValidator: &secretref.Validator{Client: client},
+	}
+
+	discoveryClient, err := discovery.NewDiscoveryClientForConfig(config.RestConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	fbReconciler := logpipelinefluentbit.New(
+		client,
+		fbConfig,
+		&workloadstatus.DaemonSetProber{Client: client},
+		flowHealthProber,
+		istiostatus.NewChecker(discoveryClient),
+		pipelineValidator,
+		&conditions.ErrorToMessageConverter{})
+
+	return fbReconciler, nil
+}
+
+//nolint:unparam // An error could be returned after implementing the IstioStatusChecker
+func configureOtelReconciler(client client.Client, config LogPipelineControllerConfig, _ *prober.LogPipelineProber) (*logpipelineotel.Reconciler, error) {
```

Comment:
I left it like that, since we will need it when implementing the SelfMonitoring for this kind of LogPipeline. If that's not desired, I can also remove it and add it later, when SM will be implemented.